lib/UserAgentApplication.ts

```diff
@@ -183,7 +170,7 @@ namespace Msal {
         * @param _tokenReceivedCallback -  The function that will get the call back once this API is completed (either successfully or with a failure).
         * @param {boolean} validateAuthority -  boolean to turn authority validation on/off.
         */
-        constructor(clientId: string, authority: string, tokenReceivedCallback: tokenReceivedCallback, validateAuthority?: boolean) {
+        constructor(clientId: string, authority: string, tokenReceivedCallback: tokenReceivedCallback, validateAuthority?: boolean, cacheLocation?: string, redirectUri?: string, postLogoutRedirectUri?: string, navigateToLoginRequestUrl?: boolean) {
```

Comment:
Why not use parameter destructuring and and default parameters for the optional parameters?